src/wallet/wallet2.cpp

```diff
@@ -13099,4 +13099,128 @@ uint64_t wallet2::get_bytes_received() const
+  m_confirm_missing_payment_id = boost::get<bool>(v[0].second);
+  rewrite(m_keys_file, pwd_container.password());
+
+  const AskPasswordType old_ask_password = m_ask_password;
+  v = {{tr("Ask password"), std::make_pair(m_ask_password, std::vector<std::pair<uint32_t, std::string>>({{AskPasswordNever, "Never"}, {AskPasswordOnAction, "Action"}, {AskPasswordToDecrypt, "Decrypt"}}))}};
+  if (!f("For security, you can set the wallet to ask for a password before sending monero and doing other operations. The 'decrypt' option also encrypts your secret spend key in memory, and your password decrypts it. 'Decrypt' is the safest option, and 'Never' is the most convenient one.", v)) return;
+  m_ask_password = (AskPasswordType)boost::get<std::pair<uint32_t, std::vector<std::pair<uint32_t, std::string>>>>(v[0].second).first;
+  if (old_ask_password == AskPasswordToDecrypt && m_ask_password != AskPasswordToDecrypt)
+    decrypt_keys(pwd_container.password());
+  else if (old_ask_password != AskPasswordToDecrypt && m_ask_password == AskPasswordToDecrypt)
+    encrypt_keys(pwd_container.password());
+  rewrite(m_keys_file, pwd_container.password());
+
+#if 0
+  // This one is probably best left out for now
+  uint32_t decimal_point = cryptonote::get_default_decimal_point();
+  v = {{tr("Unit"), std::make_pair(decimal_point, std::vector<std::pair<uint32_t, std::string>>({{12, "Monero"}, {9, "millinero"}, {6, "micronero"}, {3, "nanonero"}, {0, "piconero"}}))}};
+  if (!f("You can display amounts in a different unit. Monero is the canonical one, but you can choose to display millineros (there are 1000 of them in a Monero), etc. The unit all follow the Systeme International standard.", v)) return;
+  decimal_point = boost::get<std::pair<uint32_t, std::vector<std::pair<uint32_t, std::string>>>>(v[0].second).first;
+  cryptonote::set_default_decimal_point(decimal_point);
+  rewrite(m_keys_file, pwd_container.password());
+#endif
+
+  v = {{tr("Minimum output count"), m_min_output_count}, {tr("Minimum output value"), std::make_pair(true, m_min_output_value)}};
+  if (!f(tr("The monero wallet will usually try to consolidate outputs when you send monero. This will tend to concentrate your monero in few larger coins. If you usually need to send several transactions in short order, this can be counter productive as more of your balance will be locked waiting for confirmations. To mitigate this, you can set the wallet to avoid doing so if it would result in fewer than a certain amount of coins of a certain amount. For example, setting the count to 5 and the amount to 2 means that the wallet will try to keep at least 5 coins of 2+ monero."), v)) return;
+  m_min_output_count = boost::get<uint32_t>(v[0].second);
+  m_min_output_value = boost::get<std::pair<bool, uint64_t>>(v[1].second).second;
```

Comment:
Is there any reason why these both are collected in one invocation of the callback function? Couldn't one be grabbed, and then the other? And same thing with the case below? The primary advantage seems to be the display of the one banner message before two of the questions. Couldn't this be done by passing a `nullptr` for the input of the second value so this big banner is still only displayed once? That would drop the need for `std::vector` as input to the callback function.

Reply:
It's really one setting, composed of two values, so they're meant to be set in relation to each other.